### PR TITLE
Initial work on the opent2t reference app.

### PR DESCRIPTION
**Pull request by ChuckFerring-zz**

### CODE REVIEW

File: opent2t-reference/Opent2tHelper.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+var q = require('q');
```
Comment (shalinijoshi19) [inline opent2t-reference/Opent2tHelper.js:3]: Should this be converged with the helpers  used by teh CLI? Lots of such 'proxies' now in our code base -- maybe time to consolidate? I found myself writing one as well for the proximal case temporarily but those ended up still using the TranslatorCli/OnboardingCli helpers in the CLI #Pending

File: opent2t-reference/helpers.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        deferred.resolve(data);
+    });
+
+    return deferred.promise;
+}
+
+// logs the given error in red
+function logError(error) {
```
Comment (shalinijoshi19) [inline opent2t-reference/helpers.js]: TODO -- need to file tracking issues to get this to leverage the logging framework also ;  #Resolved

File: opent2t-reference/helpers.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        console.log(colors.red(error.stack));
+    }
+}
+
+
+function createHubDeviceFileName(hubName, deviceId) {
+    return createSaveFileName(hubName, "_device_" + deviceId);
+}
+
+function createOnboardingFileName(baseName) {
```
Comment (shalinijoshi19) [inline opent2t-reference/helpers.js]: dont we have all these in the CLI already?  #Resolved

File: opent2t-reference/main.js [hunk 4]
```diff
@@ -0,0 +1,283 @@
+var electron = require('electron');
+var app = electron.app;
+var BrowserWindow = electron.BrowserWindow;
+var q = require('q');
+var fs = require('fs');
+require('electron-reload')(__dirname);
+var Opent2tHelper = require("./Opent2tHelper");
+var opent2tHelper = new Opent2tHelper();
```
Comment (shalinijoshi19) [inline opent2t-reference/main.js:8]: is this standard (lowercase vs uppercase ) on specifying the require and the actual variable for the constructed class?  #ByDesign

File: opent2t-reference/main.js [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+
+    mainWindow.once('ready-to-show', () => {
+        mainWindow.show();
+    });
+
+    // and load the index.html of the app.
+    mainWindow.loadURL('file://' + __dirname + '/index.html');
+
+    // Open the devtools.
+    //mainWindow.openDevTools();
```
Comment (shalinijoshi19) [inline opent2t-reference/main.js]: Nit: why is this commented out? Does it need to be yanked out? if not can we have a comment explaining the comment? thanks! #Resolved

File: opent2t-reference/helpers.js (hunk 2 above)
Comment (ChuckFerring-zz) [reply opent2t-reference/helpers.js]: Removed the duplicate helpers file.

---
In reply to: [101374399](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101374399) [](ancestors = 101374399)

File: opent2t-reference/helpers.js (hunk 3 above)
Comment (ChuckFerring-zz) [reply opent2t-reference/helpers.js]: Removed the duplicate helpers file.

---
In reply to: [101374451](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101374451) [](ancestors = 101374451)

File: opent2t-reference/main.js (hunk 4 above)
Comment (ChuckFerring-zz) [reply opent2t-reference/main.js:8]: We follow this pattern in several places.

---
In reply to: [101374738](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101374738) [](ancestors = 101374738)

File: opent2t-reference/main.js (hunk 5 above)
Comment (ChuckFerring-zz) [reply opent2t-reference/main.js]: Removed commented-out code.

---
In reply to: [101374869](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101374869) [](ancestors = 101374869)

File: opent2t-reference/Opent2tHelper.js (hunk 1 above)
Comment (ChuckFerring-zz) [reply opent2t-reference/Opent2tHelper.js:3]: Yes, we should converge these.  I'd like to do that as a separate task.

---
In reply to: [101373959](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101373959) [](ancestors = 101373959)

Comment (shalinijoshi19) [reply opent2t-reference/Opent2tHelper.js:3]: Could you file a git issue/tfs track to track this? thanks a ton

---
In reply to: [101380971](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101380971) [](ancestors = 101380971,101373959)

File: opent2t-reference/main.js (hunk 4 above)
Comment (shalinijoshi19) [reply opent2t-reference/main.js:8]: k thanks

---
In reply to: [101380727](https://github.com/openT2T/opent2t-cli/pull/16#discussion_r101380727) [](ancestors = 101380727,101374738)

**Comment by shalinijoshi19** [comment]
:shipit:

**Pull request merged**